lib/ocpp/v201/init_device_model_db.cpp

```diff
@@ -0,0 +1,1508 @@
+    std::map<ComponentKey, std::vector<DeviceModelVariable>> components = get_all_component_schemas(schemas_path);
+
+    std::map<ComponentKey, std::vector<VariableAttributeKey>> component_default_values;
+    for (auto const& [componentKey, variables] : components) {
+        std::vector<VariableAttributeKey> variable_attribute_keys;
+        for (const DeviceModelVariable& variable : variables) {
+            if (variable.default_actual_value.has_value()) {
+                VariableAttributeKey key;
+                key.name = variable.name;
+                if (variable.instance.has_value()) {
+                    key.instance = variable.instance.value();
+                }
+                key.attribute_type = AttributeEnum::Actual;
+                key.value = variable.default_actual_value.value();
+                variable_attribute_keys.push_back(key);
+            }
+        }
+
+        component_default_values.insert({componentKey, variable_attribute_keys});
+    }
+
+    return component_default_values;
+}
+
+std::map<ComponentKey, std::vector<VariableAttributeKey>>
+InitDeviceModelDb::get_config_values(const std::filesystem::path& config_file_path) {
+    std::map<ComponentKey, std::vector<VariableAttributeKey>> config_values;
+    std::ifstream config_file(config_file_path);
+    json config_json = json::parse(config_file);
+    for (const auto& j : config_json.items()) {
+        ComponentKey p = j.value();
+        std::vector<VariableAttributeKey> attribute_keys;
+        for (const auto& variable : j.value().at("variables").items()) {
+            for (const auto& attributes : variable.value().at("attributes").items()) {
+                VariableAttributeKey key;
+                key.name = variable.value().at("variable_name");
+                try {
+                    key.attribute_type = conversions::string_to_attribute_enum(attributes.key());
+                } catch (const std::out_of_range& e) {
+                    EVLOG_error << "Could not find type " << attributes.key() << " of component " << p.name
+                                << " and variable " << key.name;
+                    throw InitDeviceModelDbError("Could not find type " + attributes.key() + " of component " + p.name +
+                                                 " and variable " + key.name);
+                    // TODO throw or only error??
+                }
+
+                key.attribute_type = conversions::string_to_attribute_enum(attributes.key()); // TODO try catch??
+                key.value = get_string_value_from_json(attributes.value());
+                if (variable.value().contains("instance")) {
+                    key.instance = variable.value().at("instance");
+                }
+                attribute_keys.push_back(key);
+            }
+        }
+
+        config_values.insert({p, attribute_keys});
+    }
+
+    return config_values;
+}
+
+bool InitDeviceModelDb::check_config_integrity(
```

Comment:
Should we add a check for the source of the value as well and add a log message if it is not 'default' (so the value is not changed)? And then maybe remove it from the list so we don't get more warnings about it.

Reply:
sounds like a good idea

Reply:
Hmm because it is now a vector, if there is any error in the vector it will throw. So maybe it is not the best idea after all :). I think I keep it like this.